provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -663,6 +663,20 @@ func generateSchema() schema.PackageSpec {
 						Description: "The authentication mode of the cluster. Valid values are `CONFIG_MAP`, `API` or `API_AND_CONFIG_MAP`.\n\n" +
 							"See for more details:\nhttps://docs.aws.amazon.com/eks/latest/userguide/grant-k8s-access.html#set-cam",
 					},
+					"corednsAddonOptions": {
+						TypeSpec: schema.TypeSpec{
+							Plain: true,
```

Comment:
Do you need Plain: true? I think what it does is make it slightly less nice for the user but supposedly easier for the implementation. Wanted to make sure it's an informed decision here.

Reply:
Ah yes, unsolved problem of conditional creates.

Reply:
Why does it make it less nice for the user? I thought you use plain when it doesn't need to accept output values. Is that not the case?

Reply:
Generally if I'm a user, I want to be able to call resources by passing `Output<X>` values directly. If I have plain values, great, I can lift them to outputs. If I have however computed Output values, and the resource does not accept them, I'm really stuck and I can only provision inside an apply which defeats plans. So generally Plain: true is bad for users. However the fact that this property is used for conditional resource creation inside is a good reason for having Plain: true.